### PR TITLE
Stable sort resolv.conf entries

### PR DESCRIPTION
**Pull request by dirkmueller**

### CODE REVIEW

File: chef/cookbooks/resolver/recipes/default.rb [hunk 1]
```diff
@@ -29,6 +29,7 @@
 end
 
 dns_list << node[:dns][:nameservers]
+dns_list = dns_list.flatten.sort.uniq
```
Comment (vuntz) [inline chef/cookbooks/resolver/recipes/default.rb]: I think you should sort on line 26 instead. The `uniq` here is good, though.

Comment (tboerger) [inline chef/cookbooks/resolver/recipes/default.rb]: He can put ```uniq``` even in front of ```sort```, maybe fewer entries to sort :)

Comment (tboerger) [inline chef/cookbooks/resolver/recipes/default.rb]: Even better is to append it directly a line above as the methods are not manipulating the original object.

Comment (dirkmueller) [inline chef/cookbooks/resolver/recipes/default.rb]: @vuntz I did not want to change line 26 as I want to be sure that the complete list is unified, not only a part of it.

Comment (vuntz) [inline chef/cookbooks/resolver/recipes/default.rb]: @dirkmueller but the sort is kind of wrong there, as what is in `nameservers` should definitely come after the nodes with dns-server role.

File: chef/cookbooks/resolver/recipes/default.rb [hunk 2]
```diff
@@ -24,6 +24,7 @@
 dns_list = []
 if !nodes.nil? and !nodes.empty?
   dns_list = nodes.map { |x| Chef::Recipe::Barclamp::Inventory.get_network_by_type(x, "admin").address }
+  dns_list = dns_list.flatten.sort
```
Comment (tboerger) [inline chef/cookbooks/resolver/recipes/default.rb]: What about ```dns_list.flatten!``` and ```dns_list.sort!```? That way you don't have to do it on multiple places.

**Comment by mjura** [comment]
:+1:

**Comment by vuntz** [comment]
+1

**Comment by tboerger** [comment]
:+1:

**Pull request merged**